### ISSUE TITLE
Best practice to disable a service at image run time

### ISSUE THREAD

**Issue by Luwdo** [just-containers/s6-overlay#446]
I am running using the newer s6-rc syntax for services and init scripts.

I have a number of services I want to be able to enable or disable either via environment variables or k8 config map.

Because services are not active until they are in the /etc/s6-overlay/s6-rc.d/user directory I had first thought to remove the inactive service from that directory. This can be done by running my own entry that does that before calling s6 init.

The issue with that is that if your service has dependencies you have to make sure to remove those as well or else s6 init will be waiting forever to start.

My current solution is not great but what I have done is in the run of each service i want to make "optionally" enabled I will do the following:

```
#!/command/with-contenv bash
if [ "$ENABLE_SSHD" = "yes" ]; then
  echo "[sshd] starting service"
  exec /usr/sbin/sshd -D -e
else
  echo "[sshd] sshd service disabled to enable it set ENABLE_SSHD=yes environment variable"
  exec sleep 999999
fi
```

Creating a sleep process for s6 to track but not starting my service. This has worked so far but I know there has to be a better way.

Any suggestions?

**Comment by skarnet** [comment]
s6-rc is really not made for conditional services (it was designed to reach reliable states after a reproducible startup sequence), so if you need conditional services, the way you've implemented them is pretty good.

An improvement that can be made is, instead of starting a long sleep process, you can tell the supervisor to stop the service and not restart it. Instead of `exec sleep 999999`, you can write `exec s6-svc -Od .` and it will do what you want without the extra sleep process.

**Comment by Luwdo** [comment]
Awesome I will update and test, thank you. Our use-case is a monolithic development image that some services don't need to run depending on the developers needs. We tried splitting to separate images in the same pod and what we found is that we were just guessing on resourcing for scheduling development autoscaling, we know generally we need at least x resources for the whole stack in development to run but different parts of the stack are busy at different times depending on what a developer is working on.

Our struggle for efficient resource usage in development is what lead us to this solution. Production is not an issue as every service is run in separate deployments and scaled horizontally.
